Configure SELinux in permissive mode on RL9 hosts

priteau opened the pull request:
This makes the host match the target configuration in Zed and avoids a
later reboot to change the SELinux mode.

Hosts that have SELinux disabled are rebooted only if
disable_selinux_do_reboot is changed to true.

MoteHue reviewed (dismissed):
Would be nice to recommend this in the RL9 migration docs. Otherwise, LGMT :)

markgoddard commented on etc/kayobe/inventory/group_vars/overcloud/selinux:
```diff
@@ -0,0 +1,7 @@
+
+# Do NOT reboot to apply SELinux config changes. Operators must opt-in by
+# changing this variable when applying host configuration.
+disable_selinux_do_reboot: False
```
I would prefer to change this only on RL9 hosts to avoid changing behaviour elsewhere.

markgoddard commented on etc/kayobe/inventory/group_vars/overcloud/selinux:
Should this only apply to overcloud hosts, or would it be useful to apply everywhere?

priteau replied:
I have moved it to `all` group variables which will apply to seed hypervisor, seed host and infra VMs too.

priteau committed:
Configure SELinux in permissive mode on RL9 hosts

This makes the host match the target configuration in Zed and avoids a
later reboot to change the SELinux mode.

RL9 hosts that have SELinux disabled are rebooted only if
disable_selinux_do_reboot is changed to true.

Alex-Welsh commented on etc/kayobe/inventory/group_vars/all/selinux:
```diff
@@ -0,0 +1,9 @@
+---
+# Configure SELinux in permissive mode when configuring a Rocky Linux 9 host.
+selinux_state: "{{ 'permissive' if ansible_facts.os_family == 'RedHat' and ansible_facts.distribution_major_version == '9' else 'disabled' }}"
```
Shorter but maybe less clear? What do you think?
```suggestion
selinux_state: "{{ 'disabled' if disable_selinux_do_reboot else 'permissive' }}"
```

priteau replied:
Thanks but I am not keen on this suggestion, as you have to understand the logic of the second variable to know what the first variable evaluates to.

priteau replied:
Thanks but I am not keen on this suggestion, as you have to understand the logic of the second variable to know what the first variable evaluates to.
